tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -0,0 +1,287 @@
+                                            health_obj=self.health_obj,
+                                            pod_prefix=const.SERVER_POD_NAME_PREFIX,
+                                            container_prefix=const.RGW_CONTAINER_NAME,
+                                            process=self.rgw_process, check_proc_state=False,
+                                            restart_cnt=DTM_CFG["rgw_restart_cnt"])
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+
+        self.log.info("Step 4: Wait for READ operation to complete.")
+        if proc_read_op.is_alive():
+            proc_read_op.join()
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        self.test_completed = True
+        self.log.info("ENDED: Verify READs during rgw_s3 restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-42245")
+    def test_write_during_rgw_svc_restart(self):
+        """Verify WRITEs during rgw_s3 service restart using pkill."""
+        self.log.info("STARTED: Verify WRITEs during rgw_s3 service restart using pkill")
+        log_file_prefix = 'test-42245'
+
+        que = multiprocessing.Queue()
+
+        self.log.info("Step 1: Start WRITE operation in background")
+        proc_write_op = multiprocessing.Process(target=self.dtm_obj.perform_write_op,
+                                                args=(self.bucket_name, self.object_name,
+                                                      self.test_cfg['clients'],
+                                                      self.test_cfg['samples'], log_file_prefix,
+                                                      que, self.test_cfg['size']))
+        proc_write_op.start()
+
+        self.log.info("Step 2: Perform rgw_s3 Process Restart for %s times During Read "
+                      "Operations", DTM_CFG["rgw_restart_cnt"])
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=const.SERVER_POD_NAME_PREFIX,
+                                            container_prefix=const.RGW_CONTAINER_NAME,
+                                            process=self.rgw_process, check_proc_state=False,
+                                            restart_cnt=DTM_CFG["rgw_restart_cnt"])
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+
+        self.log.info("Step 2: Wait for WRITE Operation to complete.")
+        if proc_write_op.is_alive():
+            proc_write_op.join()
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+
+        self.log.info("Step 3: Perform READ Operation on data written in Step 1")
+        self.dtm_obj.perform_ops(workload_info, que, False, True, True)
```

Comment:
can we explicitly mention no retry count once process restarted?

Reply:
So in init u have mentioned retry count and in this file ur taking it as 5 by default. So as this read verify is run after process restart is successful and no issues in write doen during the restart, was just thinking, can we rather send retry count as 0 for this particular step as retrying for 5 times after restart successful might not be reqd. Is my understanding correct? @ApurwaMohite @priyankaborawake

Reply:
@sonalk0209 you mean to perform retries only during  any process restarts and not for IOs before/after restart.